openedx/openedx-platform
# refactor: deprecate course_id from ModuleSystem [BD-13]

## Description

As a part of the ModuleSystem removal, we want to get rid of the `course_id` from its attributes. We have replaced its usages in the platform with `block.scope_ids.usage_id.context_key`. For backward compatibility, external XBlocks will still be able to implicitly obtain it either from the [CachingDescriptorSystem](https://github.com/openedx/edx-platform/blob/dd1367823d0b7fbef39511c729366ed44ea45866/xmodule/modulestore/split_mongo/caching_descriptor_system.py#L72) or [XModuleMixin](https://github.com/openedx/edx-platform/blob/b295d5a1a482b36c28ef225c9205546af6a1a2d6/xmodule/x_module.py#L381-L383) as a fallback.

## Supporting information

Sandbox instance: https://pr30715.sandbox.opencraft.hosting
Use `staff@example.com:edx` to log in.

## Testing instructions

1. Go to [the Instructor Dashboard](https://pr30715.sandbox.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-data_download) and check that generating a report of problem responses works correctly for XBLocks with custom Python scripts.
1. Check that the edX notes functionality works.
   1. First, go to [this page](https://notes.pr30715.sandbox.opencraft.hosting/) and add the certificate extension to your browser. Setting up edX notes for sandboxes needs some more steps than we have documented, but I didn't look into this. I've added `verify=False` to requests in `lms/djangoapps/edxnotes/helpers.py` to quickly get this working.
   1. Go to a page with an [HTML XBlock](https://app.pr30715.sandbox.opencraft.hosting/learning/course/course-v1:edX+DemoX+Demo_Course/block-v1:edX+DemoX+Demo_Course+type@sequential+block@19a30717eff543078a5d94ae9d6c18a5/block-v1:edX+DemoX+Demo_Course+type@vertical+block@867dddb6f55d410caaa9c1eb9c6743ec). Create, edit, and delete some notes.
   1. View all your notes [here](https://pr30715.sandbox.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/edxnotes/).
1. Go to [this page](https://app.pr30715.sandbox.opencraft.hosting/learning/course/course-v1:edX+DemoX+Demo_Course/block-v1:edX+DemoX+Demo_Course+type@sequential+block@2ba2b9bca6ef40b0bae568ee7d6fb6eb/block-v1:edX+DemoX+Demo_Course+type@vertical+block@7c5c2ce24f2a48e5b4ad774a3a3ca1b2) and check that the export functionality of the [Poll XBlock](https://github.com/open-craft/xblock-poll) is working.
1. Go through all existing XBlocks and verify that they are working correctly.
1. Check that Instructor tasks work correctly for special exams:
   1. View the course as a learner and start the [timed exam](https://app.pr30715.sandbox.opencraft.hosting/learning/course/course-v1:edX+DemoX+Demo_Course/block-v1:edX+DemoX+Demo_Course+type@sequential+block@2ba2b9bca6ef40b0bae568ee7d6fb6eb/block-v1:edX+DemoX+Demo_Course+type@vertical+block@7c5c2ce24f2a48e5b4ad774a3a3ca1b2).
   1. Refresh the page (something is wrong with the Learning MFE, as it doesn't immediately show the exam bar) and submit the exam.
   1. Switch back to the staff user and reset the attempt through the [Instructor Dashboard](https://pr30715.sandbox.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-special_exams).

## Other information

Private-ref: [BB-5969](https://tasks.opencraft.com/browse/BB-5969)
